crowd_anki/github/github_importer.py

```diff
@@ -34,21 +35,35 @@ def import_from_github(self):
         if repo and ok:
             self.download_and_import(repo)
 
-    def download_and_import(self, repo):
+    def download_and_import(self, github_repo):
+        deck_base_name = github_repo.split("/")[-1]
         try:
-            response = urllib2.urlopen(GITHUB_LINK.format(repo))
-            response_sio = StringIO.StringIO(response.read())
-            with zipfile.ZipFile(response_sio) as repo_zip:
-                repo_zip.extractall(tempfile.tempdir)
-
-            deck_base_name = repo.split("/")[-1]
-            deck_directory_wb = Path(tempfile.tempdir).joinpath(deck_base_name + "-" + BRANCH_NAME)
-            deck_directory = Path(tempfile.tempdir).joinpath(deck_base_name)
-            utils.fs_remove(deck_directory)
-            deck_directory_wb.rename(deck_directory)
-            # Todo progressbar on download
-
-            AnkiJsonImporter.import_deck(self.collection, deck_directory)
+            try:
```

Comment:
You should split this one into multiple functions (e.g. one for gitpython and one for zip method)